### PR TITLE
update plugin resource to simply work

### PR DESCRIPTION
**Pull request by Stromweld**
update plugin resource to work with newer versions of Jenkins which f…ixes dependencies and removes need for additional plugin method

Signed-off-by: Corey Hemminger <hemminger@hotmail.com>

### Description

Makes plugin lwrp useful. For latest version of plugins it'll install plugin and all dependencies via the jenkings-cli.jar install-plugin command. Newer versions of Jenkins this works great. Removes the need to brute force dependency install that would loop through all dependencies and dependencies of dependencies even if they have repeating depencies and differing versions of dependencies. 

When specifying a source or specific version dependencies are not resolved and specific plugin is installed similar to old functionality based on url or file source on node.

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by Stromweld** [comment]
This should fix #694 #653 #534 #631 #503 #494

File: kitchen.dokken.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 driver:
   name: dokken
   privileged: true # because Docker and SystemD/Upstart
-  chef_version: <%= ENV['CHEF_VERSION'] || 'current' %>
```
Comment (tas50) [inline kitchen.dokken.yml]: Was there a reason to change this. This gets us early failure detection on new chef releases, which is quite nice from a quality standpoint.

File: libraries/plugin.rb [hunk 2]
```diff
@@ -44,9 +44,6 @@ class Resource::JenkinsPlugin < Resource::LWRPBase
               default: :latest
     attribute :source,
               kind_of: String
-    attribute :install_deps,
```
Comment (tas50) [inline libraries/plugin.rb]: This should remain and instead you should throw a log warning that the attribute is deprecated and will be removed later. That way people can upgrade without breaking the world.

**Comment by tas50** [comment]
@josh-barker thoughts on this change since you did all the last work here?

File: kitchen.dokken.yml (hunk 1 above)
Comment (Stromweld) [reply kitchen.dokken.yml]: previous travis builds failed to compile runit dependency cookbook. I suspected it was due to chef-client 15.0.143 that it was using.

Comment (Stromweld) [reply kitchen.dokken.yml]: looks like I was wrong as it's still failing at the same spot. Hopefully @josh-barker may be able to point out the issue with the tests.

File: libraries/plugin.rb (hunk 2 above)
Comment (Stromweld) [reply libraries/plugin.rb]: This should be fixed now.

**Comment by Stromweld** [comment]
@josh-barker when you have some time could you look at these failing tests? Run log looks like they worked. The apache-httpcomponents plugin one I'm not sure why it didn't install the specific version, even though the plugins specifying version or source installed correct version.

**Comment by josh-barker** [comment]
Hi @stromweld, are you sure the correct version of `apache-httpcomponents` is being installed?

I spun it up and it had a newer version, not the version specified.

**Comment by Stromweld** [comment]
that was the result I got too, but if you look at the other tests that are specifying version or url and those specific versions are getting installed. Not sure why only this one doesn't.

**Comment by Stromweld** [comment]
@tas50 this should be ready for final review.

File: .travis.yml [hunk 3]
```diff
@@ -21,12 +21,12 @@ env:
   # - INSTANCE=smoke-package-stable-centos-6 # terribly slow; takes more than 45 minutes
   - INSTANCE=smoke-package-stable-centos-7
   - INSTANCE=smoke-package-stable-debian-9
-  - INSTANCE=smoke-package-stable-ubuntu-1604
```
Comment (tas50) [inline .travis.yml]: Can we get 1604 and 1804 here?

File: kitchen.dokken.yml (hunk 1 above)
Comment (tas50) [reply kitchen.dokken.yml]: runit is entirely busted at the moment (new release today maybe) in Chef 15 so this does need to be stable at the moment.

File: kitchen.dokken.yml [hunk 4]
```diff
@@ -19,6 +19,11 @@ platforms:
     image: dokken/amazonlinux
     pid_one_command: /sbin/init
 
+- name: amazonlinux-2
```
Comment (tas50) [inline kitchen.dokken.yml]: Thanks for adding this

File: kitchen.yml [hunk 5]
```diff
@@ -55,9 +61,9 @@ suites:
   - name: smoke_package_stable
     run_list: jenkins_smoke::default
     excludes:
-      - ubuntu-14.04
       - windows-2012r2
       - windows-2016
+      - widnows-2019
```
Comment (tas50) [inline kitchen.yml]: You have a typo in the exclude name here

**Comment by tas50** [comment]
Just a few minor things and this is ready to go out.

File: .travis.yml (hunk 3 above)
Comment (Stromweld) [reply .travis.yml]: yes

File: kitchen.yml (hunk 5 above)
Comment (Stromweld) [reply kitchen.yml]: nice catch

**Comment by Stromweld** [comment]
thanks for the merge

**Pull request merged**